### PR TITLE
Add Python 3.12 support

### PR DESCRIPTION
**Pull request by arkid15r**
- Use Python 3.8-3.12 for test matrix
  - Bump GH action versions
  - Fix configparser SafeConfigParser name issue (the initial one)
  - Replace pkg_resources with importlib.metadata
  - Fix failing test

### CODE REVIEW

**Review by KJhellico — APPROVED** [review]
LGTM!

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -9,15 +9,15 @@ jobs:
...
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
       - name: Set up Python ${{ matrix.python }}
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v4
```
Comment (KJhellico) [inline .github/workflows/test.yaml]:
```suggestion
        uses: actions/setup-python@v5
```

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -43,14 +43,14 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Set up Python 3.10
-        uses: actions/setup-python@v1
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v4
```
Comment (KJhellico) [inline .github/workflows/test.yaml]:
```suggestion
        uses: actions/setup-python@v5
```

Comment (arkid15r) [reply .github/workflows/test.yaml]: Done 👍

**Pull request merged**